### PR TITLE
App Ember 3.7 upgrade

### LINKED ISSUE #308: Upgrade App to ~3.7+

### PR DESCRIPTION
**Pull request by cythrawll**
Resolves #308

## Description

Upgrade App to ember 3.7! incrementally upgrade and run codemods, and get everything buttoned up and running.

## Proposed Changes

Run ember update and codemods.

### CODE REVIEW

File: packages/app/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
```
Comment (jkusa) [inline packages/app/.travis.yml]: Can we remove this?

File: packages/app/config/environment.js [hunk 2]
```diff
@@ -59,11 +58,12 @@ module.exports = function(environment) {
     ENV.APP.LOG_VIEW_LOOKUPS = false;
 
     ENV.APP.rootElement = '#ember-testing';
+    ENV.APP.autoboot = false;
   }
 
   if (environment === 'production') {
     ENV['ember-cli-mirage'] = {
-      enabled: false //set to true for gh-pages
+      enabled: true //set to true for gh-pages
```
Comment (jkusa) [inline packages/app/config/environment.js]: Is this change intentional?

File: packages/app/tests/acceptance/custom-reports-test.js [hunk 3]
```diff
@@ -1,57 +1,47 @@
...
-  assert.expect(3);
+  test('Viewing saved reports', async function(assert) {
+    assert.expect(3);
 
-  visit('/reports');
-  andThen(() => {
+    await visit('/reports');
     assert.ok(
-      Ember.isPresent(find('.navi-reports-index .navi-collection table')),
+      isPresent(find('.navi-reports-index .navi-collection table')),
```
Comment (jkusa) [inline packages/app/tests/acceptance/custom-reports-test.js]: Could this be:
```js
assert.dom('.navi-reports-index .navi-collection table').exists();
```

File: packages/app/tests/acceptance/custom-reports-test.js [hunk 4]
```diff
@@ -1,57 +1,47 @@
...
+    await visit('/reports');
+    await click(link('New Report'));
+    assert.ok(
+      currentURL().match(
+        /^\/reports\/[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}\/edit/
+      ),
+      'Clicking "New Report" button brings the user to the report builder'
+    );
+
+    assert.ok(isPresent(find('.report-builder')), 'Custom report builder is visible');
```
Comment (jkusa) [inline packages/app/tests/acceptance/custom-reports-test.js]: same here

File: packages/app/config/environment.js (hunk 2 above)
Comment (cythrawll) [reply packages/app/config/environment.js]: see comment, wondering why it was ever set to false.

File: packages/app/.travis.yml (hunk 1 above)
Comment (cythrawll) [reply packages/app/.travis.yml]: the entire file?

File: packages/app/tests/helpers/contains-helpers.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import { findAll } from '@ember/test-helpers';
```
Comment (jkusa) [inline packages/app/tests/helpers/contains-helpers.js]: Should we put this in core so other packages can use this?

File: packages/app/config/environment.js (hunk 2 above)
Comment (jkusa) [reply packages/app/config/environment.js]: For the `gh-pages` demo we need to enable the mock for `prod` builds manually. However, as a user of the Navi app, when you clone the repo and configure your web services, you would want to disable mirage for prod builds.

File: packages/app/.travis.yml (hunk 1 above)
Comment (jkusa) [reply packages/app/.travis.yml]: Yes, we use the top level .travis.yml file.

Comment (cythrawll) [reply packages/app/.travis.yml]: we didn't remove navi-data one.  Also, do we really want to remove this on every ember-upgrade?

File: packages/app/app/services/navi-notifications.js [hunk 6]
```diff
@@ -34,7 +34,7 @@ export default Ember.Service.extend({
    * @returns {Ember.Service} notification service
    */
   add(options = {}) {
-    let notificationService = get(this, 'notificationService');
+    let notificationService = this.notificationService;
```
Comment (mattgraham1995) [inline packages/app/app/services/navi-notifications.js]: Why did this change?

File: packages/app/app/services/navi-notifications.js [hunk 7]
```diff
@@ -56,6 +56,6 @@ export default Ember.Service.extend({
    * @returns {Ember.Service} navi notification service
    */
   clearMessages() {
-    return get(this, 'notificationService').clearMessages();
+    return this.notificationService.clearMessages();
```
Comment (mattgraham1995) [inline packages/app/app/services/navi-notifications.js]: Same here

File: packages/core/test-support/helpers/contains-helpers.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import { findAll } from '@ember/test-helpers';
+
+/**
+ * Finds selector by innerText
+ * @param {String} selector - selector to search
+ * @param {String} text - text to search by
+ * @returns {Element} - first matching dom element
+ */
+export default function findByContains(selector, text) {
+  return findAll(selector).filter(el => el.innerText.includes(text))[0];
```
Comment (mattgraham1995) [inline packages/core/test-support/helpers/contains-helpers.js]: We could just use `find` instead of filter and then we wouldn't need the `[0]` at the end

File: packages/core/test-support/helpers/contains-helpers.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+/**
+ * Finds button by innnerText
+ * @param {String} text - text to search for
+ * @returns {Element} - first matching dom element
+ */
+export function button(text) {
```
Comment (mattgraham1995) [inline packages/core/test-support/helpers/contains-helpers.js]: Maybe `button` and `link` would be better with more descriptive names. They make sense in the scope of this file, but anywhere it's imported to wouldn't necessarily have the context to tell what these functions do.

File: packages/core/test-support/helpers/contains-helpers.js (hunk 8 above)
Comment (mattgraham1995) [reply packages/core/test-support/helpers/contains-helpers.js]: Also what does this return if the element isn't found? Would an ` || null;` at the end be necessary?

File: packages/app/app/services/navi-notifications.js (hunk 6 above)
Comment (cythrawll) [reply packages/app/app/services/navi-notifications.js]: gets are no longer needed.

File: packages/app/app/services/navi-notifications.js (hunk 7 above)
Comment (cythrawll) [reply packages/app/app/services/navi-notifications.js]: same

File: packages/core/test-support/helpers/contains-helpers.js (hunk 8 above)
Comment (cythrawll) [reply packages/core/test-support/helpers/contains-helpers.js]: we need all the dom elements so the filter works.

also it would return undefined which is appropriate I think.

Comment (mattgraham1995) [reply packages/core/test-support/helpers/contains-helpers.js]: But we're only returning the first one that matches the filter which is exactly what find does
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Comment (mattgraham1995) [reply packages/core/test-support/helpers/contains-helpers.js]: `return findAll(selector).find(el => el.innerText.includes(text));`
